design-proposals/volume-update-strategy.md

```diff
@@ -0,0 +1,218 @@
+# Overview
+This design proposal extends KubeVirt APIs to enable the storage migration between PVCs.
+
+## Motivation
+In some circumstances, the guest owner may want to have the option of switching the storage provider or class for a group of volumes used by an active VM. Currently, the VM must be turned off and recreated, causing service disruptions and downtime.
+
+
+## Goals
+Design and develop an API to live update and migrate PVCs.
+  * The volume migration needs to be declarative and [GitOps](https://kubevirt.io/user-guide/operations/gitops/) compatible
+  * Volume migration implies an update of the volumes used by the VM. Hence, we want to trigger the storage migration on a VM update.
+  * The VM spec updates needs to be triggered by the user and not to be implicitly updated by the KubeVirt controller
+
+ ## Non  Goals
+*  This feature is not directly aimed at live migrating the VM using RWO volumes. Hence, the API covers and refers only to the volume update.
```

Comment:
This is actually the first I'm hearing of this limitation.  But I guess I'm not surprised.  What will happen if user creates a RWO volume by mistake?

Reply:
@mhenriks maybe the sentence isn't clear. I don't think this isn't something new. What I wanted to say is that this feature doesn't enable general VM live-migration with RWO volumes. If your VM uses RWO volumes and the user want to live migrate (standard one, with the VMI migration creation object) then they still get an error message that the volume needs to use RWX mode.
This feature only cover the case when you want to change PVC not just migrating the VM

Reply:
I tried to be more explicit here, please, check if now the sentence is more clear